Packs/Bitbucket/Integrations/Bitbucket/Bitbucket.py

```diff
@@ -0,0 +1,1330 @@
+        return self._http_request(method='GET', url_suffix=url_suffix, params=params)
+
+    def pull_request_comment_update_request(self, repo: str, pr_id: int, body: Dict, comment_id: str) -> Dict:
+        """ Makes a PUT request /repositories/workspace/repository/pullrequests/{pr_id}/comments/{comment_id} endpoint
+            to update a specific comment in a pull request.
+            :param repo: str - The repository the user entered, if he did.
+            :param pr_id: str - an id to a specific pull.
+            :param body: Dict - a dictionary with the updated content of the comment.
+            :param comment_id: str - an id to a specific comment to update.
+
+            Creates the url and makes the api call
+            :return JSON response from /repositories/workspace/repository/pullrequests/{pr_id}/comments/{comment_id} endpoint
+            :rtype Dict[str, Any]
+        """
+        url_suffix = f'/repositories/{self.workspace}/{repo}/pullrequests/{pr_id}/comments/{comment_id}'
+        return self._http_request(method='PUT', url_suffix=url_suffix, json_data=body)
+
+    def pull_request_comment_delete_request(self, repo: str, pr_id: int, comment_id: str) -> Response:
+        """ Makes a DELETE request /repositories/workspace/repository/pullrequests/{pr_id}/comments/{comment_id} endpoint
+            to delete a specific comment in a pull request.
+            :param repo: str - The repository the user entered, if he did.
+            :param pr_id: str - an id to a specific pull request to add a comment to.
+            :param comment_id: str - an id to a specific comment to delete.
+
+            Creates the url and makes the api call
+            :return JSON response from /repositories/workspace/repository/pullrequests/{pr_id}/comments/{comment_id} endpoint
+            :rtype Dict[str, Any]
+        """
+        url_suffix = f'/repositories/{self.workspace}/{repo}/pullrequests/{pr_id}/comments/{comment_id}'
+        return self._http_request(method='DELETE', url_suffix=url_suffix, resp_type='response')
+
+    def workspace_member_list_request(self, params: Dict) -> Dict:
+        """ Makes a GET request /workspaces/{workspace}/members endpoint
+            to return a list of the members in the workspace.
+            :param params: Dict - The pagination params if needed
+
+            Creates the url and makes the api call
+            :return JSON response from /workspaces/{workspace}/members endpoint
+            :rtype Dict[str, Any]
+        """
+        url_suffix = f'/workspaces/{self.workspace}/members'
+        return self._http_request(method='GET', url_suffix=url_suffix, params=params)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def check_pagination(client: Client, response: Dict, limit: int) -> List:
+    arr: List[Dict] = response.get('values', [])
+    is_next = response.get('next', None)
+    pagelen = response.get('pagelen', None)
+    if is_next and pagelen and limit > int(pagelen):
```

Comment:
Can you explain this?

Reply:
* The field is_next contains the api call to the next page of results, if exists. In the response it is under the key "next".
* The field pagelen contains the number of results in the page, or the maximun results that can be in a page if it is not given as a parameter in the API request. 
* The response from the API returns as a dictionary that contains a list of the results (with "values" as the key), pagelen, size (The number of results overall), page and next. 
* The if statement checks if there is a url to the next page of results (is_next) and if there is a value in pagelen, and if there is, is it smaller than the number required in the limit?
* If there is a url to the next page and pagelen < limit, which means that there are more results and the client asked for more records than there is in this API response, then we need to to a pagination, and make another api call, to get more results.
An example for is_next = https://api.bitbucket.org/2.0/workspaces/{self.workspace}/projects?pagelen=1&page=2.
An example for a scenario is: if the maximum number of results in the API response is 20, but the client wants 50 results, then in the response there will be a next field, like above, and limit > int(pagelen), so we will go to the pagination function.

Reply:
But we don't need to check if there is a pagelen, no?
It is set by the limit or by default. It will always exist